### PR TITLE
PIM-6536: rework history grid

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

Rwork of the history grid
**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | NA
| Added Behats                      | NA
| Added integration tests           | NA
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/cell/expand-cell.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        return Backgrid.StringCell.extend({
+            /** @property */
+            className: "AknGrid-bodyCell AknGrid-expandable",
+
+            /**
+             * Render the completeness.
+             */
+            render: function () {
+                this.$el.empty().html(
+                    '<span class="AknButtonList"><div class="AknButtonList-item version-expander AknGrid-expand"></div><span class="AknButtonList-item AknButton AknButton--grey AknButton--round">' + this.model.get(this.column.get('name')) + '</span></span>'
```
Comment (tamarasaurus) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/cell/expand-cell.js]: you could stick this in a html file

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/history-row.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        versionTemplate: _.template(versionTemplate),
+        changesTemplate: _.template(changesTemplate),
+
+        /**
+         Renders a row of cells for this row's model.
+        */
+        render: function () {
+            this.$el.empty();
+
+            var mainLine = $(this.versionTemplate({
```
Comment (tamarasaurus) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/history-row.js]: const

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/history-row.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            }));
+
+            mainLine.on('click', function () {
+                changesLine.toggle();
+                mainLine.toggleClass('AknGrid-bodyRow--expanded')
+                mainLine.find('.AknGrid-expand').toggleClass('AknGrid-expand--expanded')
+            });
+
+            for (var i = 0; i < this.cells.length; i++) {
+                var cell = this.cells[i];
```
Comment (tamarasaurus) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/history-row.js]: can you explain what's happening here, not sure I understand

File: src/Pim/Bundle/EnrichBundle/Resources/config/datagrid/history.yml [hunk 4]
```diff
@@ -7,11 +7,12 @@ datagrid:
         options:
             requireJSModules:
                 - oro/datagrid/pagination-input
+            rowView: oro/datagrid/history-row
```
Comment (tamarasaurus) [inline src/Pim/Bundle/EnrichBundle/Resources/config/datagrid/history.yml]: cool

File: src/Pim/Bundle/DataGridBundle/Extension/Formatter/Property/ChangesetProperty.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
+<?php
+
```
Comment (SamirBoulil) [inline src/Pim/Bundle/DataGridBundle/Extension/Formatter/Property/ChangesetProperty.php]: declare strict type + typehints

Comment (juliensnz) [reply src/Pim/Bundle/DataGridBundle/Extension/Formatter/Property/ChangesetProperty.php]: I cannot: it inherit from another class

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/cell/expand-history-cell.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+/* global define */
+define(['underscore', 'backgrid', 'pim/template/datagrid/cell/expand-history-cell'],
+    function(_, Backgrid, template) {
+        'use strict';
+
+        /**
+         * Number column cell.
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/cell/expand-history-cell.js]: copy / paste ?

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/cell/history-diff-cell.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+/* global define */
+define(['oro/translator', 'backgrid', 'pim/template/datagrid/cell/history-diff-cell'],
+function(__, Backgrid, template) {
+    'use strict';
+
+    /**
+     * String column cell. Added missing behaviour.
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/cell/history-diff-cell.js]: idem

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/history-row.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+     *
+     * Triggers events:
+     *  - "clicked" when row is clicked
+     *
+     * @export  oro/datagrid/row
+     * @class   oro.datagrid.Row
+     * @extends Backgrid.Row
+     */
+    return BaseRow.extend({
+        tagName: 'div',
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/history-row.js]: Isn't it the default value ?

File: src/Pim/Bundle/DataGridBundle/Resources/public/lib/backgrid/backgrid.js [hunk 9]
```diff
@@ -2234,7 +2235,14 @@ var Body = Backgrid.Body = Backbone.View.extend({
     var fragment = document.createDocumentFragment();
     for (var i = 0; i < this.rows.length; i++) {
       var row = this.rows[i];
-      fragment.appendChild(row.render().el);
+      row.render();
+      if ('TR' === row.el.tagName) {
+        fragment.appendChild(row.render().el);
```
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/lib/backgrid/backgrid.js]: Why do you call render() a second time ?

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/history-row.js (hunk 8 above)
Comment (juliensnz) [reply src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/history-row.js]: yes but I override so I need to specify it :)

**Pull request merged**